### ISSUE TITLE
'Unstage file' stopped working.

### ISSUE THREAD

**Issue by allsorts46** [gitextensions/gitextensions#8007]
Since 'recently' (sorry, can't recall exactly when), the single-file 'unstage' button in the commit dialog has stopped working - if I stage a file I can not unstage it again. The only way is to use the unstage all files button and start again.

I am not sure how this is implemented, but in case this is helpful to you: trying to unstage the file myself in git bash I found that 'git restore [file]' also did absolutely nothing, but with the added 'staged' switch, as in 'git restore [file] --staged' did do what I wanted. Not sure if this is something that has changed.

## Current behaviour

Cannot unstage individual file in commit window. File remains in bottom pane.

## Expected behaviour

'Unstage' button returns selected file to the top 'Working directory changes' pane.

## Steps to reproduce

1. Stage a file
2. Try to unstage the file
3. Nothing happens, stays staged. No errors or feedback of any kind.

## Did this work in previous version of GitExtensions

Yes, but I do not know when exactly this started, sorry.

## Environment

- Git Extensions 3.3.1.7897
- Build 5a97671645532bcedc443bac7b727f40db47cb5c
- Git 2.26.1.windows.1
- Microsoft Windows NT 10.0.18362.0
- .NET Framework 4.8.4121.0
- DPI 96dpi (no scaling)

**Comment by pmiossec** [comment]
I see that you have the last version of git. Could it be linked to the update?

**Comment by RussKie** [comment]
Has it started after you upgraded the app? Or has it been working in the v3.3.1 and then suddenly stopped?
If it is the former, we'll need more info to investigate. If it is the latter - then there is something else is to blame, like git.

**Comment by gerhardol** [comment]
There were corrections to quoting in 3.3, like #7288 Some further changes in #7615 
If this is GE, there could be because of the Git clone path, specific filename or specific locale
Please try master portable builds, soon to be 4.0

**Comment by allsorts46** [comment]
I can no longer intentionally reproduce this, even though I haven't changed either my version of Git or GitExtensions. I have been active, switching branches a few times, etc.

Given that when I tried 'git restore' on the command line myself nothing happened but no error was given, I assume that GitExtensions was in a similar position - it requested the file to be restored, the command completed with no error, and so it assumed success even though nothing had changed. If there is a bug here, seems more likely it's in Git.

I did have a colleague experience exactly the same issue, also using a fresh install of the latest versions of Git and GitExtensions, and on the same codebase.

If either of us can find a reliable way to cause it I'll post back, otherwise I guess it makes sense to close this issue until some more concrete steps are found.

Thanks for the suggestions.
